### DISCUSSION TITLE
ImageMagick convert from ai to png shows some patches in the image

### DISCUSSION THREAD

**Discussion by 9shet56** [ImageMagick/ImageMagick#4241] [Help]
### ImageMagick version

ImageMagick 7.0.10-3 Q16

### Operating system

Windows

### Operating system, version and so on

Windows 2016 Standard, 64 bit OS

### Description

We are using an Adobe Illustrator type of Image and trying to convert into png using ImageMagick. After converting it shows up some extra patches in the image in our Production environment, but the same image in our non production environment shows up correctly without any black patches.
The ImageMagick version in both the environments are same.

'Im unable to attach the file to this issue.

### Steps to Reproduce

Used below command in the server having ImageMagick.
("
magick convert <filename.ai> <filename.png>
")

**Comment by 9shet56** [comment]
Any other way can I attach the error screenshots?

**Comment by dlemstra** [comment]
You are most likely using a different version of Ghostscript in both situations.

**Comment by cancan101** [comment]
Does ImageMagick support opening `.ai` (Adobe Illustrator) files by way of Ghostscript? I didn't see that format listed at: https://imagemagick.org/script/formats.php

It looks like yes, `magick convert` will take an input `ai` file. In the case of the file I tested (CMYK space) the colors did get a bit messed up.

**Reply by fmw42** [reply]
In IM 7 use magick, not magick convert.  I suspect Imagemagick uses Ghostscript to open ai files similarly to pdf and ps files.

**Comment by snibgo** [comment]
@cancan101: Yes, IM delegates rasterization of AI files to Ghostscript. Inserting "-verbose" at the start will show this. Yes, https://imagemagick.org/script/formats.php should list AI but doesn't. This a documentation bug.
